tokio-sync/src/barrier.rs

```diff
@@ -0,0 +1,117 @@
+use crate::watch;
+use crate::Mutex;
```

Comment:
What is the reasoning to use `crate::Mutex` vs. `std::sync::Mutex`?

Reply:
Not sure I follow? `std::sync::Mutex` isn't async. We need to use an async-aware `Mutex` here.

Reply:
We chatted on Gitter, but for the record:

In theory, `std::sync::Mutex` should be faster than `tokio::sync::Mutex` when:

- The critical section is small
- There are no `.await` calls within the critical section.